api/src/main/scala/hmda/api/http/institutions/submissions/RequestVerificationUtils.scala

```diff
@@ -61,20 +60,20 @@ trait RequestVerificationUtils extends HmdaCustomDirectives {
     } yield msg
   }
 
-  private def getErrorMessage(i: Institution, f: Filing, s: Submission, iid: String, p: String, sid: SubmissionId)(implicit ec: ExecutionContext): Future[Option[String]] = Future {
+  private def getErrorMessage(i: Option[Institution], f: Filing, s: Submission, iid: String, p: String, sid: SubmissionId)(implicit ec: ExecutionContext): Future[Option[String]] = Future {
     val submissionFound: Boolean = s.id == sid
     val filingFound: Boolean = f.period == p
-    val institutionFound: Boolean = i.id == iid
+    val institutionFound: Boolean = i.getOrElse(Institution.empty).id == iid
```

Comment:
We still have to compare the two institution IDs, so I'm not sure if the `isDefined` method would help here.  Can you give me an example of what you were thinking of?

Reply:
I thought we only had to compare the two institution ids to see whether the found Institution was a real institution or the "empty Institution". Now that `GetInstitutionById` returns `None` instead of "empty Institution", there's a much easier way to check whether the method call was successful:

```scala
val institutionFound: Boolean = i.isDefined
```

This is the kind of change it would be nice to make all over the codebase, to make all of our operations more typesafe.

Reply:
Beware of using `isDefined` for the `Option` type. It's not good idiomatic `Scala` in most code bases, and there are better alternatives (i.e. pattern matching, `find` + `getOrElse`, the `collect` method, etc). This [article](http://danielwestheide.com/blog/2012/12/19/the-neophytes-guide-to-scala-part-5-the-option-type.html) has some arguments and examples of this.

Reply:
Okay. It's not worth arguing about this small example, so it's fine the way it is. There are definitely ways we can make it better Scala if we decide to refactor in the future.